message_definitions/v1.0/ras_a.xml

```diff
@@ -16,6 +16,67 @@
         <param index="2" label="X Position in the camera frame" minValue="0" increment="1">Position given by the number of pixels in X direction of the camera frame. Origin on the bottom left corner of the camera frame.</param>
         <param index="3" label="Y Position in the camera frame" minValue="0" increment="1">Position given by the number of pixels in Y direction of the camera frame. Origin on the bottom left corner of the camera frame.</param>
       </entry>
+      <entry value="36" name="MAV_CMD_DO_FOLLOW_SYS_ID" hasLocation="false" isDestination="false">
+        <description>Begin following a target</description>
+        <param index="1" label="System ID" minValue="1" maxValue="255" increment="1">System ID (of the FOLLOW_TARGET beacon).</param>
+        <param index="2">Reserved</param>
+        <param index="3">Reserved</param>
+        <param index="4" label="Altitude Mode" minValue="0" maxValue="2" increment="1">Altitude mode: 0: Keep current altitude, 1: keep altitude difference to target, 2: go to a fixed altitude above home.</param>
+        <param index="5" label="Altitude" units="m">Altitude above home. (used if mode=2)</param>
+        <param index="6">Reserved</param>
+        <param index="7" label="Time to Land" units="s" minValue="0">Time to land in which the MAV should go to the default position hold mode after a message RX timeout.</param>
+      </entry>
+      <entry value="37" name="MAV_CMD_DO_FOLLOW_POI" hasLocation="false" isDestination="false">
+        <description>Begin following a target specified by POI ID.  A POI Unique ID (UID) is a 64-bit field, and so to fully represent that UID, the big endian 64-bit POI UID is split into 2 words; param1: upper 32-bit word, param2: lower 32-bit word.</description>
+        <param index="1" label="POI UID Upper 32 Bits">The upper 32 bits of the UID from a POI_REPORT.</param>
+        <param index="2" label="POI UID Lower 32 Bits">The lower 32 bits of the UID from a POI_REPORT</param>
+        <param index="3" label="Mode Flags" enum="FOLLOW_POI_MODE_FLAGS">Bitmask of option flags.</param>
+        <param index="4" label="Altitude" units="mm or degE7">Altitude field, behavior to be interpreted according to Mode Flags.</param>
+        <param index="5" label="Azimuth" units="degE7">Azimuth field, behavior to be interpreted according to Mode Flags.</param>
+        <param index="6" label="Range" units="mm">Range field, behavior to be interpreted according to Mode Flags.</param>
+        <param index="7" label="Time to Return" units="s" minValue="0">Time to return in which the Autopilot should enter return to comms behavior. If 0, vehicle will immediaately enter return to comms behavior after losing comms with GCS</param>
```

Comment:
Really need a separate TTR time to the normal data loss timeouts? This would also imply that Follow me without a GCS cannot be supported, which is a bit rubbish. 
Probably should have a value for "just default to system behaviour for data loss" if this is really needed.

Are there any cases where any of these values might have defaults or might not be supplied? Usually we'd use a value to indicate "ignore/use default/use current?

Reply:
not sure why this implies follow me cannot be supported without GCS?

Reply:
>> Time to return in which the Autopilot should enter return to comms behavior.

> not sure why this implies follow me cannot be supported without GCS?

G'day @thomas-watters-skydio ,

A data timeout implies that you have data connection, which implies that you have a GCS (or telemetry connection of some kind). So if you have no GCS you would have to set a very large timeout to stop this immediately returning.

ON that note, you should specify a way to indicate no timeout - usually NaN.

> this TTR should be separate from default system TTR, we are more like more comfortable following a subject longer then we would be for example flying under manual control?

The default system TTR will trigger irrespective of what you do for this value, if it is enabled. So this can only make the system trigger earlier than the default system timeout. It also seems to me that returning is just one possible action. Check out what PX4 offers generally for safety failsafes: http://docs.px4.io/master/en/config/safety.html#safety-configuration-failsafes

If this was for use in a mission then I would propose a command for setting the action on data loss in the mission - much like [MAV_CMD_SET_FENCE_BREACH_ACTION](https://mavlink.io/en/messages/development.html#MAV_CMD_SET_FENCE_BREACH_ACTION) (not implemented). 


As it is not, I think that perhaps it is reasonable to have a command to set the system defaults. You then set/reset them value as needed for your system at the time. This is a good idea anyway because it is something you would reasonably want to do through a GCS and you don't want to have to mess with parameters to do.